admin-ui/src/main/java/io/crate/rest/action/admin/AdminUIFrontpageAction.java

```diff
@@ -84,4 +99,106 @@ private boolean isBrowser(String headerValue) {
 
         return matcher.matches();
     }
+
+    private class AdminUIRequestFilter extends RestFilter {
+
+        @Override
+        public void process(RestRequest request, RestChannel channel, RestFilterChain filterChain) throws IOException {
+            if(request.rawPath().startsWith(ADMIN_ENPOINT)) {
+                serveSite(request, channel);
+            } else {
+                filterChain.continueProcessing(request, channel);
+            }
+        }
+    }
+
+    private void serveSite(RestRequest request, RestChannel channel) throws IOException {
+        if (request.method() != RestRequest.Method.GET) {
```

Comment:
would be faster to check for GET first and then for the admin endpoint imho ( so I'd move this check to AdminUIRequestFilter.process())